packages/connect/src/utils/firmwareReleaseConfigUtils.ts

```diff
@@ -6,122 +6,119 @@ import { getJWSPublicKey } from '@trezor/env-utils';
-        }
-        if (localJws.sequence >= decodedJws.payload.sequence) {
-            // If we fetch remote but local is newer or equal to remote then we use local.
-            decodedJws = decodedJwsLocal;
-        }
-    }
+    return JSON.parse(decoded.payload);
+};
 
-    const algorithmInHeader = decodedJws?.header.alg;
-    if (algorithmInHeader !== JWS_SIGN_ALGORITHM) {
-        throw Error(`Wrong algorithm in JWS config header: ${algorithmInHeader}`);
+const verifyJwsSignature = (jws: string, publicKey: string): void => {
+    const decoded = decode(jws);
+    if (decoded?.header.alg !== JWS_CONFIG.SIGN_ALGORITHM) {
+        throw new Error(
+            `Invalid JWS algorithm: expected ${JWS_CONFIG.SIGN_ALGORITHM}, got ${decoded?.header.alg}`,
+        );
     }
 
-    const JWSPublicKey = getJWSPublicKey(
-        'firmware-release',
-        // When using local we always want to use CodeSignKey.
-        ['test-signed', 'production'].includes(env) || !isRemote,
-    );
-    if (!JWSPublicKey) {
-        throw new Error('JWS public key is not defined!');
+    if (!verify(jws, JWS_CONFIG.SIGN_ALGORITHM, publicKey)) {
+        throw new Error('JWS signature is invalid.');
     }
+};
 
-    try {
-        const isAuthenticityValid = verify(releasesJws, JWS_SIGN_ALGORITHM, JWSPublicKey);
-        if (!isAuthenticityValid) {
-            throw new Error('Config authenticity is invalid');
+export const getFirmwareReleaseConfig = async () => {
+    const { jws: remoteJws, source: initialSource, env } = await fetchRemoteJWS();
+
+    let finalJws = remoteJws;
+    let finalSource = initialSource;
+
+    if (initialSource === 'remote') {
+        try {
+            const remotePayload = decodeJwsPayload(remoteJws);
+            const localPayload = decodeJwsPayload(firmwareReleaseConfigAssets.jws);
+
+            // Sanity check local config version.
+            if (localPayload.version !== JWS_CONFIG.VERSION) {
+                throw new Error(`Local config has mismatched version: ${localPayload.version}`);
+            }
+
+            // If local is newer or same sequence, prefer it over remote.
+            if (localPayload.sequence >= remotePayload.sequence) {
+                finalJws = firmwareReleaseConfigAssets.jws;
+                finalSource = 'local';
+            }
+        } catch (error) {
+            console.error(`Error comparing remote/local JWS: ${error}. Using remote as is.`);
         }
-
-        return JSON.parse(decodedJws.payload);
-    } catch (error) {
-        console.error('Error validating:', error);
-        throw new Error(`Failed to validate release message: ${error.message}`);
     }
-};
 
-export const getFirmwareReleaseConfig = async () => {
-    const { releasesJws, isRemote, env } = await getReleaseJWS();
-    const config: FirmwareReleaseConfig = verifyFirmwareRelease(releasesJws, isRemote, env);
+    const useProductionKey = ['test-signed', 'production'].includes(env) || finalSource === 'local';
```

Comment:
I'd find this a bit easier to read:
```suggestion
    const useProductionKey = env === 'test-signed' || env === 'production' || finalSource === 'local';
```